cql3/statements/select_statement.cc

```diff
@@ -543,14 +544,16 @@ indexed_table_select_statement::do_execute_base_query(
         auto &merger = query_state.merger;
         auto &keys = query_state.primary_keys;
         auto &key_it = query_state.current_primary_key;
-        return repeat([this, &keys, &key_it, &merger, &proxy, &state, &options, cmd, timeout]() {
+        return repeat([this, &keys, &key_it, &merger, &proxy, &state, &options, cmd, timeout, &query_state]() {
             // Starting with 1 key, we check if the result was a short read, and if not,
             // we continue exponentially, asking for 2x more key than before
```

Comment:
*** I also edited the example implementation by changing `keys.begin() + 100` to `key_it + 100`, I'm writing it just in case GitHub doesn't notify when a comment is edited.

Reply:
Yes, it looks like the author of the patch can't read.

We have exponentials without cutoff elsewhere too, you aren't the only one guilty of it. And in fact it's hard to think of a good cutoff, low ones will result in slow processing on large clusters that _can_ handle the concurrency.

I'll adjust the code to use std::min() as you suggest.

Reply:
The author of the comment above can't read either, you also told me about the other places. :man_facepalming: